src/packet_analysis/protocol/ip/AnalyzerAdapter.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include "zeek/analyzer/Analyzer.h"
+
+namespace zeek::analyzer::pia { class PIA; }
+
+namespace zeek::packet_analysis::IP {
+
+class IPBasedAnalyzer;
+
+/**
+ * This class represents the interface between the packet analysis framework and
+ * the session analysis framework. One of these should be implemented for each
+ * packet analyzer that intends to forward into the session analysis.
+ */
+class AnalyzerAdapter : public analyzer::Analyzer {
```

Comment:
sorry but one more naming idea for this: how about `SessionAdaptor`? that would make the tie in with the session framework apparent.

Reply:
Seems reasonable to me. I'll squash that back into the `Adapter` commit before I merge this.